kmeta/owner_references_test.go

```diff
@@ -46,19 +49,40 @@ func TestNewControllerRef(t *testing.T) {
 		},
 	}
 
-	blockOwnerDeletion := true
-	isController := true
 	want := &metav1.OwnerReference{
 		APIVersion:         "example.knative.dev/v1alpha1",
 		Kind:               "Frobber",
 		Name:               "foo",
 		UID:                "42",
-		BlockOwnerDeletion: &blockOwnerDeletion,
-		Controller:         &isController,
+		BlockOwnerDeletion: ptr.Bool(true),
+		Controller:         ptr.Bool(true),
 	}
 
 	got := NewControllerRef(f)
 	if diff := cmp.Diff(want, got); diff != "" {
 		t.Error("Unexpected OwnerReference (-want +got):", diff)
 	}
 }
+
+func TestOwnerRefableDeployment(t *testing.T) {
+	d := &appsv1.Deployment{
+		TypeMeta: metav1.TypeMeta{},
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "foo",
+			UID:  "42",
+		},
+	}
+	want := &metav1.OwnerReference{
+		APIVersion:         "apps/v1",
+		Kind:               "Deployment",
+		Name:               "foo",
+		UID:                "42",
+		BlockOwnerDeletion: ptr.Bool(true),
+		Controller:         ptr.Bool(true),
```

Comment:
the deployment is the controller of whatever you are making? This seems strange to me at first read, likely you are not reconciling deployments and I will bet you want more than one owner on the child object? 

What is the usecase?

Reply:
https://github.com/knative/eventing/pull/4731/files#diff-3d78ac5bf206c11ebf4dae96f827bdb6a30b4d50a03c12a1bd65ee1af73782c2R167